x/evm/keeper/state_transition.go

```diff
@@ -140,13 +140,15 @@ func (k *Keeper) ApplyTransaction(tx *ethtypes.Transaction) (*types.MsgEthereumT
 	txHash := tx.Hash()
 	res.Hash = txHash.Hex()
 
-	logs := k.GetTxLogs(txHash)
-	res.Logs = types.NewLogsFromEth(logs)
-
-	// update block bloom filter
-	bloom := k.GetBlockBloomTransient()
-	bloom.Or(bloom, big.NewInt(0).SetBytes(ethtypes.LogsBloom(logs)))
-	k.SetBlockBloomTransient(bloom)
+	// We need to ignore the log in case transaction is reverted
+	if !res.Reverted {
```

Comment:
If you don't set a cache context, the state for a reverted tx will still be committed: 

```go
cacheCtx, commit := ctx.CacheContext()
k.ctx = cacheCtx

res, err := k.ApplyMessage(evm, msg, ethCfg)
// error handling

// set logs as previously

// commit changes to the store
if !res.Reverted {
    commit()
}
```

Reply:
ok, got it.

In that case we need to refactor a bit more. The refund gas is executed in applyMessage. we should still commit that even in case of revert

Reply:
fixed